core/src/main/java/org/bitcoinj/crypto/KeyCrypterAESCBC.java

```diff
@@ -115,6 +117,31 @@ EncryptedData encrypt(byte[] plainBytes, byte [] iv, KeyParameter aesKey) throws
         }
     }
 
+    @Override
+    public Protos.Wallet.EncryptionType getUnderstoodEncryptionType() {
+        return Protos.Wallet.EncryptionType.ENCRYPTED_BLS_KEYEXCHANGE_AES;
+    }
+
+    @Override
+    public KeyParameter deriveKey(CharSequence password) throws KeyCrypterException {
+        throw new UnsupportedOperationException("use one other deriveKey methods instead");
+    }
```

Comment:
This class and derived classes do not support deriving a key from a password.  Instead this class, keys are "derived" from private keys of different formats.

So the keys are not derived, rather the byte array is wrapped inside a KeyParameter object.

Derived classes can derive a key through other means.